src/client-private.js

```diff
@@ -183,9 +183,23 @@ function onPendingSession (sessionInfo) {
     address: sessionInfo.fromJid.split('@')[0],
     conversationId: sessionInfo.conversationId
   };
+
+  const existingSessionId = Object.keys(this._pendingSessions).find(k => {
+    const session = this._pendingSessions[k];
+    return session && session.conversationId === sessionInfo.conversationId;
+  });
+
+  if (existingSessionId) {
+    this._log('info', 'duplicate session invitation, ignoring', sessionInfo);
+    return;
+  }
+
   this.emit('pendingSession', sessionEvent);
 
   this._pendingSessions[sessionEvent.id] = sessionEvent;
+  setTimeout(() => {
```

Comment:
Why are we clearing the pending session from the list after 1 second?

Reply:
The idea here is that if you don't handle it within a second, it'll re-alert by adding it as a new pending session. Also important because you could decline, and it could get sent back to you as a new session. Previously we didn't track them with state at all, now we do, but only for 1 second intervals to debounce it, effectively.